zerver/lib/test_runner.py

```diff
@@ -104,6 +106,18 @@ def run_test(test: TestCase, result: TestResult) -> bool:
 
     test_name = full_test_name(test)
 
+    # Write each test being run to a file classified by its PID.
+    # This allows reference to the set of tests ran in a failed worker.
+    fn = "var/test-backend-%s.json" % (os.getpid(),)
+    try:
+        with open(fn) as json:
+            file = ujson.load(json)
+    except IOError:
+        file = []
+    file.append(test_name)
+    with open(fn, 'w') as json:
+        ujson.dump(file, json)
```

Comment:
I'm confused by this log structure -- if we're just appending individual items, why not just have them one per line, rather than using JSON?  Seems a lot simpler.